### PR TITLE
Added linkedin auth

### PR DESCRIPTION
**Pull request by Dhroov7**

### CODE REVIEW

File: config.js [hunk 1]
```diff
@@ -4,8 +4,9 @@ config = {
   "FACEBOOK_LOGIN_SCOPES": ["email", "public_profile"],
   "TWITTER_CALLBACK": "/login/twitter/callback",
   "GITHUB_CALLBACK": "/login/github/callback",
-  "GOOGLE_CALLBACK": "/login/google/callback",  
-  "GRANT_TOKEN_SIZE": 32,
+  "GOOGLE_CALLBACK": "/login/google/callback",
+    "LINKEDIN_CALLBACK": "/connect/linkedin/callback",
```
Comment (championswimmer) [inline config.js]: indentation

File: src/db/models.js [hunk 2]
```diff
@@ -176,7 +180,7 @@ if (!process.env.ONEAUTH_DB_NO_SYNC) {
 
 module.exports = {
     models: {
-        User, UserLocal, UserFacebook, UserTwitter, UserGithub,UserGoogle, UserLms,
+        User, UserLocal, UserFacebook, UserTwitter, UserGithub,UserGoogle,UserLinkedin, UserLms,
```
Comment (championswimmer) [inline src/db/models.js]: space after comma, always

File: src/routers/connect/linkedin.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+            failureRedirect: '/users/me',
+            failureFlash: true
+        })(req, res, next)
+    }
+}
+
+router.get('/', passport.authorize('linkedin'))
+
+
+router.get('/callback',authnOrAuthzLinkedin ,function (req,res) {
```
Comment (championswimmer) [inline src/routers/connect/linkedin.js]: shouldn't be authnOrAuthz, it should only always be authorize, never authenticate as it is not used for login, but only for connecting.

File: src/routers/disconnect/linkedin.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
+const router = require('express').Router()
+const models = require('../../db/models').models
+
+function DisconnectLinkedin(req, res) {
+
+    let existingUser = req.user
+
+    if (!existingUser) {
+
+        res.redirect('/')
```
Comment (championswimmer) [inline src/routers/disconnect/linkedin.js]: should have a error flash describing that you can only connect existing accounts, not create new one

**Comment by Dhroov7** [comment]
@championswimmer Done with the changes and also resolve the conflicts.

File: src/routers/connect/linkedin.js [hunk 5]
```diff
@@ -4,28 +4,19 @@ const passport = require('../../passport/passporthandler')
...
             failureFlash: true
         })(req, res, next)
-    }
 }
 
 router.get('/', passport.authorize('linkedin'))
 
 
-router.get('/callback',authnOrAuthzLinkedin ,function (req,res) {
+router.get('/callback',AuthzLinkedin ,function (req,res) {
```
Comment (championswimmer) [inline src/routers/connect/linkedin.js]: just put passport.authorize here no? the additional functional wrapper was there to do the if-else for `req.isAuthenticated`, which is not needed anymore now. Also function names start with smallcase. Been quite a while I am nitpicking this, please make it a habit.

File: config.js [hunk 6]
```diff
@@ -1,59 +1,56 @@
-
 config = {
-  "FACEBOOK_CALLBACK": "/login/facebook/callback",
```
Comment (championswimmer) [inline config.js]: please do not commit all these whitespace changes

**Comment by Dhroov7** [comment]
@championswimmer Done.

**Comment by Dhroov7** [comment]
@championswimmer Removed all the white spaces.

**Comment by Dhroov7** [comment]
@championswimmer Resolved Conflicts.

**Pull request closed without merging**